src/components/Inngest.ts

```diff
@@ -490,8 +490,31 @@ const builtInMiddleware = (<T extends MiddlewareStack>(m: T): T => m)([
     name: "Inngest: Logger",
     init({ client }) {
       return {
-        onFunctionRun() {
-          const logger = new ProxyLogger(client["logger"]);
+        onFunctionRun(arg) {
+          const { ctx } = arg;
+          const metadata = {
+            runID: ctx.runId,
+            eventName: ctx.event.name,
+            functionName: arg.fn.name,
+          };
+
+          let providedLogger: Logger = client["logger"];
+          /* eslint-disable @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/ban-ts-comment */
+          // create a child logger if the provided logger has child logger implementation
+          try {
+            if ("child" in client["logger"]) {
+              type ChildLoggerFn = (
+                metadata: Record<string, unknown>
+              ) => Logger;
+              // @ts-ignore-2339: ignore child method is not defined error since it's already checked above
```

Comment:
Nit - I think you can remove all of these `@ts-ignore` and `eslint-disable` comments if you check `if ("child" in providedLogger)` instead of `client["logger"]`.